src/app/utils/field-validation/validators/checkboxValidator.ts

```diff
@@ -0,0 +1,141 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+
+import { ProcessedCheckboxResponse } from 'src/app/modules/submission/submission.types'
+import { ICheckboxField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+
+type CheckboxValidator = ResponseValidator<ProcessedCheckboxResponse>
+type CheckboxValidatorConstructor = (
+  checkboxField: ICheckboxField,
+) => CheckboxValidator
+
+/**
+ * Returns a validator to check if answerArray is empty
+ */
+const checkboxAnswerValidator: CheckboxValidator = (response) => {
+  const { answerArray } = response
+
+  return answerArray.length === 0
+    ? left(`CheckboxValidator:\t Answer is empty array`)
+    : right(response)
+}
+
+/**
+ * Returns a validation function to check if number of
+ * selected checkbox options is less than the minimum number specified.
+ */
+const minOptionsValidator: CheckboxValidatorConstructor = (checkboxField) => (
+  response,
+) => {
+  const { validateByValue } = checkboxField
+  const { customMin } = checkboxField.ValidationOptions
+  const { answerArray } = response
+
+  return validateByValue && customMin
```

Comment:
instead of using a nested ternary operator here, can we just return early using `if (!validateByValue || !customMin) return right(response)`?